src/backend/base/langflow/components/langchain_utilities/SpiderTool.py

```diff
@@ -0,0 +1,129 @@
+            "display_name": "Depth",
+            "field_type": "int",
+            "info": "The crawl limit for maximum depth. If 0, no limit will be applied.",
+            "advanced": True,
+        },
+        "blacklist": {
+            "display_name": "Blacklist",
+            "field_type": "str",
+            "info": "Blacklist paths that you do not want to crawl. Use Regex patterns.",
+            "advanced": True,
+        },
+        "whitelist": {
+            "display_name": "Whitelist",
+            "field_type": "str",
+            "info": "Whitelist paths that you want to crawl, ignoring all other routes. Use Regex patterns.",
+            "advanced": True,
+        },
+        "use_readability": {
+            "display_name": "Use Readability",
+            "field_type": "bool",
+            "info": "Use readability to pre-process the content for reading.",
+            "advanced": True,
+        },
+        "request_timeout": {
+            "display_name": "Request Timeout",
+            "field_type": "int",
+            "info": "Timeout for the request in seconds.",
+            "advanced": True,
+        },
+        "metadata": {
+            "display_name": "Metadata",
+            "field_type": "bool",
+            "info": "Include metadata in the response.",
+            "advanced": True,
+        },
+        "params": {
+            "display_name": "Additional Parameters",
+            "field_type": "dict",
+            "info": "Additional parameters to pass to the API. If provided, other inputs will be ignored.",
+        },
+    }
+
+    def build(
+        self,
+        spider_api_key: str,
+        url: str,
+        mode: str,
+        limit: Optional[int] = 0,
+        depth: Optional[int] = 0,
+        blacklist: Optional[str] = None,
+        whitelist: Optional[str] = None,
+        use_readability: Optional[bool] = False,
+        request_timeout: Optional[int] = 30,
+        metadata: Optional[bool] = False,
+        params: Optional[Data] = None,
+    ) -> Data:
+        if params:
+            parameters = params.__dict__["data"]
+        else:
+            parameters = {
+                "limit": limit,
+                "depth": depth,
+                "blacklist": blacklist,
+                "whitelist": whitelist,
+                "use_readability": use_readability,
+                "request_timeout": request_timeout,
+                "metadata": metadata,
+                "return_format": "markdown",
+            }
+
+        app = Spider(api_key=spider_api_key)
+        try:
+            if mode == "scrape":
+                parameters["limit"] = 1
+                result = app.scrape_url(url, parameters)
+            elif mode == "crawl":
+                result = app.crawl_url(url, parameters)
+            else:
+                raise ValueError(f"Invalid mode: {mode}. Must be 'scrape' or 'crawl'.")
+        except Exception as e:
+            raise Exception(f"Error: {str(e)}")
+
+        records = []
```

Comment:
This returns a linting error: 
![image](https://github.com/langflow-ai/langflow/assets/131612909/a40f77e2-55a2-47f3-8e7f-824d9a886be1)

But I would argue that it is the right return type because the window where you can see what the tool returned is so big and allows for multiple entries.